bitbots_test/docs/manual/writing_tests.rst

```diff
@@ -0,0 +1,182 @@
+Writing Tests
+=============
+This guide describes how new test cases can be created as well as some best practices to follow.
+
+Most packages which use *bitbots_test* will have a :file:`test/` folder in which all test code resides.
+This folder is further divided between the different kinds of tests that *bitbots_test* supports.
+Currently there are two kinds of tests:
+
+:Unit Tests:
+
+   Unit tests are designed to extensively test small components or units of a larger system.
+   *bitbots_test* supports unit tests written in :ref:`Python <writing_unit_tests_in_python>` as well as ones written in
+   :ref:`C++ <writing_unit_tests_in_cxx>`.
+   They are located (by default) in :file:`test/unit_tests`.
+
+:rostests:
+
+   rostests are basically launch files which contain one or more ``<test>`` tags.
+   They can be used for package tests that require interacting with ROS as well as integration or system tests.
+   *bitbots_test* only supports :ref:`writing rostests <writing_rostests>` in Python.
+   They are located (by default) in :file:`test/rostests`.
+
+
+.. _writing_unit_tests_in_python:
+
+Writing Unit-Tests in Python
+----------------------------
+
+#) Add a file like :file:`test/unit_tests/test_{something}.py`.
+   These test files should not be executable.
+
+#) Create one or more classes derived from :class:`bitbots_test.test_case.TestCase` in that file.
+
+#) Create one ore more methods in your class prefixed with ``test_``.
+   When executing tests, each of these methods will be called as a separate test.
+
+   .. seealso:: :doc:`./executing_tests` on how to do the executing.
+
+#) Call one of the provided ``assert_`` methods defined on your derived class.
```

Comment:
Is there a resource for the available assert methods? Also, they do not contain a `_`.

Reply:
fixed in 71dd94fa1b6c0378f8d239416bf24a72b08578a7